kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java

```diff
@@ -112,6 +113,9 @@ public class JSONMarshaller implements Marshaller {
 
     private boolean findDeserializerFirst = Boolean.parseBoolean(System.getProperty("org.kie.server.json.findDeserializerFirst.enabled", "true"));
 
+    private boolean typeFactoryPerMarshaller = Boolean.parseBoolean(System.getProperty("org.kie.server.json.typeFactoryPerMarshaller.enabled", "true"));
```

Comment:
I wondered whether "true" or "false" is better as a default value.

Pros for "true" as default:
* Less memory retention especially for frequent deploy/dispose operations.

Pros for "false" as default:
* singleton `_typeFactory._typeCache` is a limited sized cache so it's not an infinite memory leak.
* Shared cache may reduce the heap size in some scenarios (e.g. many different deployments)

Considering the above, hmm, I vote for "true" as default.